### DISCUSSION TITLE
Difficulties when frequency or growth rate crosses zero

### DISCUSSION THREAD

**Discussion by DustinSch** [danielver02/ALPS#171] [Q&A]
Hello,
I have successfully replicated the results of your paper. For further testing purposes, I would like to reproduce the oblique proton and electron firehouse instability as described by Maneva et al. 2016 ApJ (Fig.1).
However, my ALPS setup consistently fails when the real part or imaginary part crosses zero, as this causes the corresponding solution to erroneously stay zero or to have the wrong sign.

The results for two propagation angles (where one can see that the imaginary part stays zero once its becomes zero and that the frequency has the wrong sign) are included in the appendix, along with the input file.
In Section 3.1 of your paper, you describe the method for integration near poles. Changing these parameters (or the generell resolution in momentum or k-space), I have not been able to resolve the issue so far (I also get no error output).
Do you have any suggestions regarding which parameters I should modify and how?

[ofhi_35deg_omega_worng_sign_gamma_stays_zero.pdf](https://github.com/danielver02/ALPS/files/12597572/ofhi_35deg_omega_worng_sign_gamma_stays_zero.pdf)
[ofhi_15deg_gamma_stays_zero.pdf](https://github.com/danielver02/ALPS/files/12597578/ofhi_15deg_gamma_stays_zero.pdf)
[test_ofhi_15deg.log](https://github.com/danielver02/ALPS/files/12597630/test_ofhi_15deg.log)

**Comment by kgklein** [comment]
Dustin-
My first suggestion for cases such as these are to adjust the wavevector resolution to coarser or finer values, but it appears you have already tried this. How significantly did you change the wavevector resolution, and are you running in a linear or log wavevector scan? Sometimes a log scan will more easily stay on the 'correct' mode.

My other initial suggestion would be to look at a complex frequency map of the solutions just above and just below the solution jump, and identify the correct modes you would like to follow, and perform a wavevector scan to larger and smaller values, splicing together these solutions. 

Cheers
Kris

**Reply by DustinSch** [reply]
Thank you for your prompt response!
I have already experimented with different wavevector grids, coarse ones (ns=25, ni=1) and fine ones (ns=1000, ni=1, and ns=500, ni=10, etc.). And I used the log wavevector scan.
One of the primary challenges I encounter is to find the desired mode when the initial guess value is close to zero. In my attempts to replicate certain electron instabilities, I have renormalized the input parameters, namely the guess of omega, gamma and the wave numbers to the electron scale. Despite employing the map scan method, I still only obtain a solution where both omega and gamma remain zero across the entire wavevector scan, even with a high-resolution wavevector grid.

The same issue persists with the case I previously uploaded: When I move immediately after/before the solution jump (with gamma close to zero and omega=0) and use the logarithmic map scan, ALPS only detects and traces the „solution“ where both the real and the imaginary parts remain zero (for both scans to lower and higher wavenumbers).
